checkout.c

```diff
@@ -42,3 +47,59 @@ const char *unique_tracking_name(const char *name, struct object_id *oid)
 	free(cb_data.dst_ref);
 	return NULL;
 }
+
+int detach_head_to(struct object_id *oid, const char *action,
+		   const char *reflog_message)
```

Comment:
The commit message could tell the new function name to make it easier for reviewer.

For example:

"The motivation behind this commit is to extract the core part of do_reset() from `sequencer.c` and move it to a new detach_head_to() function in `checkout.c`."

Reply:
This is good advice.